src/query-builder/replace-query-builder.ts

```diff
@@ -0,0 +1,373 @@
+import { OperationNodeSource } from '../operation-node/operation-node-source.js'
```

Comment:
Couldn't you use `InsertQueryBuilder` instead of creating this copy-pasted builder? It probably has features that are not supported by MySQL replace queries but I think that's less bad than having another query builder to maintain. You don't need to change the `InsertQueryBuilder's` documentation either. Simply an example in the `replaceInto` methods documentation is enough.

Reply:
`replace into` is probably used rarely. We want to minimise the impact of features that are hardly ever used.

Reply:
I agree, this is a niche feature that is not worth all the additional complexity added in my first attempt.